test/unit/core/action-creator.ts

```diff
@@ -155,20 +156,203 @@ suite('ActionCreator', ({ expect, spy, stub }) => {
       it('should return an action with validation if search term is different', () => {
         const search: any = { a: 'b' };
         const state = { a: 'b' };
-        stub(Selectors, 'query').withArgs(state).returns('book');
+        stub(Selectors, 'query').returns('book');
+        stub(Selectors, 'selectedRefinements').returns([]);
 
         expectAction(() => actions.updateSearch(search), Actions.UPDATE_SEARCH, search,
           (meta) => expect(meta.validator.payload[1].func({ query: 'boot' }, state)).to.be.true);
       });
 
-      it('should return an action with validation if search term is null', () => {
+      it('should return an action with validation that evaluates to ' +
+        'false if search term is null and refinements are same', () => {
         const query = null;
```

Comment:
we should try to pull as many of these `const`s as possible into the containing `describe()` to simplify each `it()`s setup

Reply:
I'm thinking `state`, `query`, `search`, `refinement`